experimental/wgit/weigit_api.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import os
+import pathlib
+import sys
+
+import pygit2
+
+
+class WeiGit:
+    def __init__(self) -> None:
+        """
+        Planned Features:
+            1. create the wgit directory. Error, if already dir exists.
+            2. SHA1Store.init()
+            3. Create SHA1 .wgit/sha1_ref_count.json
+            3. Initialize a .git directory within the .wgit using `git init`.
+            4. add a .gitignore within the .wgit directory, so that the git repo within will ignore `sha1_ref_count.json`
+        """
+
+        # Make .wgit directory
+        try:
+            os.mkdir(".wgit")
+        except FileExistsError:
+            sys.stderr.write("WeiGit has been already Initialized \n")
+            sys.exit(0)
+
+        # if no .wgit dir then initialize the following
+        SHA1_store()
+
+        # create sha1_ref_count
+        with open(".wgit/sha1_ref_count.json", "w") as f:
+            print("The sha1_ref_count.json file has been created!")
+
+        # Make the .wgit a git repo, add a gitignore and add SHA1_ref
+        pygit2.init_repository(".wgit/.git", False)
+        with open("./.wgit/.gitignore", "w") as f:
```

Comment:
there should be a check that this file is not already there?

Reply:
I think for this one, this code snippet only executes if `.wgit` doesn't exist (if .wgit exists then we error out in line 30). Should still checking be better?

Reply:
True. There could be a rare race condition that after the first check, this file shows up. But you are right, we don't need the check here. Can you please add a comment here?

Reply:
Yes adding a comment to reflect it.

Reply:
Adding a try and except, in case we face the race condition.